src/cGame_logic.cpp

```diff
@@ -2180,4 +2180,65 @@ void cGame::onKeyDownDebugMode(const cKeyboardEvent &event)
 void cGame::setMousePosition(int w, int h)
 {
     m_mouse->setCursorPosition(window, w,h);
-}
\ No newline at end of file
+}
+
+void cGame::execute(AbstractMentat &mentat)
+{
+    if (game.isState(GAME_BRIEFING)) {
```

Comment:
it would be even more cool if this was somewhere provided when constructing the mentat class. Ie, in case of campaign mission inject behavior to get to the next mission. Else, for skirmish do other things. Wdyt? (what do you think)

Reply:
I remember putting this bunch of code here because the lambda function was getting complex to write and I didn't want to rewrite everything, because with the future state branch, this code would no longer have its place here and would be completely rewritten. Let's say it's temporary. 
I'm not very familiar with how the software knows where the player is in the missions, but it's true that an eventHandler would be more judicious.